### PR TITLE
Add query ID to Session and ConnectorSession

### PR DESCRIPTION
**Pull request by haozhun**

### CODE REVIEW

**Comment by cberner** [comment]
Looks like the tests are failing

**Comment by haozhun** [comment]
Tests are fixed.

**Comment by electrum** [comment]
Title this "query ID"

File: presto-main/src/main/java/com/facebook/presto/FullConnectorSession.java [hunk 1]
```diff
@@ -39,11 +40,13 @@
     private final SessionPropertyManager sessionPropertyManager;
 
     public FullConnectorSession(
+            String queryId,
             Identity identity,
             TimeZoneKey timeZoneKey,
             Locale locale,
             long startTime)
     {
+        this.queryId = queryId;
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/FullConnectorSession.java]: Check for null

File: presto-main/src/main/java/com/facebook/presto/FullConnectorSession.java [hunk 2]
```diff
@@ -30,6 +30,7 @@
 public class FullConnectorSession
         implements ConnectorSession
 {
+    private final String queryId;
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/FullConnectorSession.java]: Add this in `toString()`

File: presto-main/src/main/java/com/facebook/presto/FullConnectorSession.java [hunk 3]
```diff
@@ -63,6 +67,7 @@ public FullConnectorSession(
             String catalog,
             SessionPropertyManager sessionPropertyManager)
     {
+        this.queryId = queryId;
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/FullConnectorSession.java]: check for null here too

File: presto-main/src/main/java/com/facebook/presto/Session.java [hunk 4]
```diff
@@ -37,6 +38,7 @@
 
 public final class Session
 {
+    private final QueryId queryId;
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/Session.java]: add to `toString()`

**Comment by cberner** [comment]
Ask @dain or @martint to look at the SPI change, but otherwise it looks good to me

File: presto-main/src/main/java/com/facebook/presto/server/ResourceUtil.java [hunk 5]
```diff
@@ -128,6 +129,8 @@ public static Session createSessionForRequest(HttpServletRequest servletRequest,
             }
         }
 
+        sessionBuilder.setQueryId(queryId);
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/server/ResourceUtil.java]: This should be set above, where `SessionBuilder` is created

File: presto-spi/src/test/java/com/facebook/presto/spi/block/TestDictionaryBlockEncoding.java [hunk 6]
```diff
@@ -38,6 +38,12 @@
     private static final ConnectorSession SESSION = new ConnectorSession()
     {
         @Override
+        public String getQueryId()
+        {
+            return "test_dictblock_query_id";
```
Comment (electrum) [inline presto-spi/src/test/java/com/facebook/presto/spi/block/TestDictionaryBlockEncoding.java]: Use `"test_connector_query_id"`, so it is clear that the value isn't special

File: presto-spi/src/test/java/com/facebook/presto/spi/block/TestVariableWidthBlockEncoding.java [hunk 7]
```diff
@@ -35,6 +35,12 @@
     private static final ConnectorSession SESSION = new ConnectorSession()
     {
         @Override
+        public String getQueryId()
+        {
+            return "test_varblock_query_id";
```
Comment (electrum) [inline presto-spi/src/test/java/com/facebook/presto/spi/block/TestVariableWidthBlockEncoding.java]: Use `"test_connector_query_id"`

File: presto-tests/src/test/java/com/facebook/presto/execution/TestQueues.java [hunk 8]
```diff
@@ -170,4 +159,22 @@ private static DistributedQueryRunner createQueryRunner(Map<String, String> prop
...
         }
     }
+
+    private static Session newSession()
+    {
+        return testSessionBuilder()
+                .setCatalog("tpch")
+                .setSchema("sf100000")
+                .build();
+
```
Comment (electrum) [inline presto-tests/src/test/java/com/facebook/presto/execution/TestQueues.java]: Extra blank line

**Comment by haozhun** [comment]
@dain If you are too busy to take a look, please assign it to @martint. Thanks.

**Comment by dain** [comment]
This change seems reasonable to me.

**Pull request merged**